cli/src/cli_util.rs

```diff
@@ -1814,10 +1837,19 @@ to the current parents may contain changes from multiple commits.
                 self.write_commit_summary(formatter, &commit)?;
                 Ok(())
             });
-            error.add_hint(
-                "Pass `--ignore-immutable` or configure the set of immutable commits via \
-                 `revset-aliases.immutable_heads()`.",
-            );
+            error.add_hint("Immutable commits are used to protect shared history.");
+            error.add_hint(indoc::indoc! {"
+                For more information, see:
+                      - https://jj-vcs.github.io/jj/latest/config/#set-of-immutable-commits
+                      - `jj help -k config`, \"Set of immutable commits\""});
+
+            let exact = upper_bound == Some(lower_bound);
+            let or_more = if exact { "" } else { " or more" };
+            let s = if exact && lower_bound == 1 { "" } else { "s" };
+            error.add_hint(format!(
+                "This operation would rewrite {lower_bound}{or_more} immutable commit{s}."
+            ));
```

Comment:
nit: I prefer to simplify it and always use plural. That's what we generally do. I don't think "rewrite 1 commits" looks strange, perhaps because it's written with a numeric "1" (not "one"), or maybe it's because I'm used to that from Mercurial (I'm also not a native English speaker but my language also has plurals).

Gluing together strings like this is bad for i18n, if we ever want to support that.